src/LegacyStrings.jl

```diff
@@ -49,15 +49,15 @@ using Compat
     end
 
     if VERSION >= v"0.5.0-"
-        immutable ASCIIString <: DirectIndexString
+        struct ASCIIString <: DirectIndexString
```

Comment:
SALSA.jl uses `UTF32String` for some reason, I can't see why, since it is not for interop.

SQLite.jl uses `UTF16String` for calling external C interfaces, but it looks like it could use `Base.transcode` instead.